roles/ceph-rgw/tasks/multisite/main.yml

```diff
@@ -2,27 +2,57 @@
 - name: include multisite checks
   include_tasks: checks.yml
 
+- name: add realm to rgw stanza in ceph.conf
+  ini_file:
+    dest: "/etc/ceph/{{ cluster }}.conf"
+    section: "client.rgw.{{ ansible_hostname }}.{{ item.instance_name }}"
+    option: "rgw_realm"
+    value: "{{ rgw_realm }}"
+  with_items: "{{ rgw_instances }}"
+  when: rgw_instances is defined
+
+- name: add zonegroup to rgw stanza in ceph.conf
+  ini_file:
+    dest: "/etc/ceph/{{ cluster }}.conf"
+    section: "client.rgw.{{ ansible_hostname }}.{{ item.instance_name }}"
+    option: "rgw_zonegroup"
+    value: "{{ rgw_zonegroup }}"
+  with_items: "{{ rgw_instances }}"
+  when: rgw_instances is defined
+
+- name: add zonegroup to rgw stanza in ceph.conf
+  ini_file:
+    dest: "/etc/ceph/{{ cluster }}.conf"
+    section: "client.rgw.{{ ansible_hostname }}.{{ item.instance_name }}"
+    option: "rgw_zone"
+    value: "{{ rgw_zone }}"
+  with_items: "{{ rgw_instances }}"
+  when: rgw_instances is defined
+
+- name: include_tasks create_realm_zonegroup_zone_lists.yml
+  include_tasks: create_realm_zonegroup_zone_lists.yml
+
 # Include the tasks depending on the zone type
 - name: include_tasks master.yml
   include_tasks: master.yml
   when:
     - rgw_zonemaster | bool
     - not rgw_zonesecondary | bool
 
+- name: include_tasks start_radosgw.yml for zonemaster rgws
+  include_tasks: ../start_radosgw.yml
+  when:
+    - rgw_zonemaster | bool
+    - not rgw_zonesecondary | bool
+
 - name: include_tasks secondary.yml
   include_tasks: secondary.yml
   when:
     - not rgw_zonemaster | bool
     - rgw_zonesecondary | bool
 
-# Continue with common tasks
-- name: add zone to rgw stanza in ceph.conf
-  ini_file:
-    dest: "/etc/ceph/{{ cluster }}.conf"
-    section: "client.rgw.{{ ansible_hostname }}.{{ item.instance_name }}"
-    option: "rgw_zone"
-    value: "{{ rgw_zone }}"
-  with_items: "{{ rgw_instances }}"
+- name: include_tasks start_radosgw.yml for zonesecondary rgws
+  include_tasks: ../start_radosgw.yml
   when:
-    - rgw_instances is defined
-  notify: restart rgw
+    - not rgw_zonemaster | bool
+    - rgw_zonesecondary | bool
```

Comment:
Why are you including the start_radosgw.yml file again for both master and secondary task when this should have been done during the "classic" ceph-rgw role ?

Reply:
So @cbodley pointed out to be that no rgws should be turned on until after the period has been updated. I've done that for both the secondary and master zone rgws now. Also the classic ceph-rgw role still applies because that call to start_radosgw.yml applies when `rgw_multisite: false`.